src/HtmlParser.js

```diff
@@ -50,7 +51,9 @@ module.exports = class HtmlParser extends Transform {
       while (i < cacheLen){
         c = cache.charCodeAt(i)
         if (s === TEXT){
-          if (c === 32 || c >= 9 && c <= 13) // ws
+          if ( c === 32 && !preserveNbsp)
```

Comment:
Based on my previous comment, if we treat all whitespace the same, this would change to:

`if (!preserveWhitespace && (c === 32 || c >= 9 && c <= 13)){...}`

and we would get rid of the next `else if` condition

Reply:
The idea of removing all `\r` and `\n` is really appealing since they *almost* never have an effect on how a browser displays the markup. It compensates for templating frameworks, e.g. handlebars, that keep indentation when rendering html (unless you sprinkle your templates with tildes :().

On the other hand then the lib should treat [pre-tags](https://github.com/paed01/atlas-html-stream/tree/pre-tag) and whatnots that expects unaltered text. It may result in degraded performance.

I guess your suggestion is the way to go!?

Reply:
No, I agree with you, spaces are important so we'll treat them specially. I'll merge this soon, then push `1.1.0` as a minor release. It's 4:00am for me, so I might do this tomorrow.